src/ambulance_game/markov/waiting.py

```diff
@@ -303,56 +297,63 @@ def mean_waiting_time_formula(
     threshold : int
     system_capacity : int
     buffer_capacity : int
-    formula : str, optional
+    waiting_formula : function
 
     Returns
     -------
     float
-        The mean waiting time for the specified class
+        The overall mean waiting time by combining class 1 and class 2 individuals
     """
-    if formula == "recursive":
-        mean_waiting_time = mean_waiting_time_formula_using_recursive_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
-
-    if formula == "algebraic":
-        mean_waiting_time = mean_waiting_time_formula_using_algebraic_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
+    mean_waiting_time_class_1 = waiting_formula(
+        all_states=all_states,
+        pi=pi,
+        class_type=1,
+        lambda_2=lambda_2,
+        lambda_1=lambda_1,
+        mu=mu,
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        buffer_capacity=buffer_capacity,
+    )
+    mean_waiting_time_class_2 = waiting_formula(
+        all_states=all_states,
+        pi=pi,
+        class_type=2,
+        lambda_2=lambda_2,
+        lambda_1=lambda_1,
+        mu=mu,
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        buffer_capacity=buffer_capacity,
+    )
 
-    if formula == "closed_form":
-        mean_waiting_time = mean_waiting_time_formula_using_closed_form_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
+    prob_accept_class_1 = np.sum(
+        [
+            pi[state]
+            for state in all_states
+            if is_accepting_state(state, 1, threshold, system_capacity, buffer_capacity)
```

Comment:
Same approach here (and also let's add named vars to the `is_accepting_state` call.